cannon/mipsevm/exec/mips_instructions.go

```diff
@@ -343,48 +343,35 @@ func ExecuteMipsInstruction(insn uint32, opcode uint32, fun uint32, rs, rt, mem
 		case 0x0F: // lui
 			return SignExtend(rt<<16, 32)
 		case 0x20: // lb
-			msb := uint32(arch.WordSize - 8) // 24 for 32-bit and 56 for 64-bit
-			return SignExtend((mem>>(msb-uint32(rs&arch.ExtMask)*8))&0xFF, 8)
+			return SelectSubWord(rs, mem, 1, true)
```

Comment:
Shouldn't we port these methods into `MIPSInstructions.sol`?

Reply:
Yes. Will do.

Reply:
Done.